Datadog.Serverless.Compat/DatadogServerlessCompat.cs

```diff
@@ -0,0 +1,184 @@
+        private static readonly string? BinaryPath = Environment.GetEnvironmentVariable(
+            "DD_SERVERLESS_COMPAT_PATH"
+        );
+        private static readonly ILogger _logger;
+
+        static DatadogServerlessCompat()
+        {
+            var logLevelEnv = Environment.GetEnvironmentVariable("DD_LOG_LEVEL");
+
+            var logLevelMapping = new Dictionary<string, LogLevel>(StringComparer.OrdinalIgnoreCase)
+            {
+                { "OFF", LogLevel.None },
+                { "CRITICAL", LogLevel.Critical },
+                { "ERROR", LogLevel.Error },
+                { "WARN", LogLevel.Warning },
+                { "INFO", LogLevel.Information },
+                { "DEBUG", LogLevel.Debug },
+                { "TRACE", LogLevel.Trace }
+            };
+
+            LogLevel logLevel = LogLevel.Information;
+            if (
+                !string.IsNullOrEmpty(logLevelEnv)
+                && logLevelMapping.TryGetValue(logLevelEnv, out var mappedLevel)
+            )
+            {
+                logLevel = mappedLevel;
+            }
+
+            var loggerFactory = LoggerFactory.Create(builder =>
+            {
+                builder.AddConsole().SetMinimumLevel(logLevel);
+            });
+            _logger = loggerFactory.CreateLogger("Datadog.Serverless.Compat");
+        }
+
+        private static bool IsWindows()
+        {
+            return RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+        }
+
+        private static bool IsLinux()
+        {
+            return RuntimeInformation.IsOSPlatform(OSPlatform.Linux);
+        }
+
+        private static CloudEnvironment GetEnvironment()
+        {
+            var env = Environment.GetEnvironmentVariables();
+
+            if (
+                env.Contains("FUNCTIONS_EXTENSION_VERSION")
+                && env.Contains("FUNCTIONS_WORKER_RUNTIME")
+            )
+            {
+                return CloudEnvironment.AzureFunction;
+            }
+
+            if (env.Contains("FUNCTION_NAME") && env.Contains("GCP_PROJECT"))
+            {
+                return CloudEnvironment.GoogleCloudRunFunction1stGen;
+            }
+
+            return CloudEnvironment.Unknown;
+        }
+
+        private static string GetBinaryPath()
+        {
+            if (!string.IsNullOrEmpty(BinaryPath))
+            {
+                _logger.LogDebug("Detected user configured binary path {BinaryPath}", BinaryPath);
+                return BinaryPath;
+            }
+
+            if (IsWindows())
+            {
+                _logger.LogDebug("Detected {OS}", OS);
+                return "datadog/bin/windows-amd64/datadog-serverless-compat.exe";
+            }
+            else
+            {
+                _logger.LogDebug("Detected {OS}", OS);
+                return "datadog/bin/linux-amd64/datadog-serverless-compat";
+            }
+        }
+
+        private static string SetPackageVersion()
```

Comment:
Should this be called `GetPackageVersion`?

Reply:
Yes I think that makes sense. Updated here: https://github.com/DataDog/datadog-serverless-compat-dotnet/commit/85ad5daf0e26e9ae43a732593f6c7853ccee6bdb